addons/web_editor/static/src/js/editor/rte.summernote.js

```diff
@@ -35,106 +35,23 @@ var tplButton = renderer.getTemplate().button;
 var tplIconButton = renderer.getTemplate().iconButton;
 var tplDropdown = renderer.getTemplate().dropdown;
 
-function _rgbToHex(cssColor) {
-    var rgba = cssColor.match(/^rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*(\d+(?:\.\d+)?))?\)$/);
-    if (!rgba) {
-        return cssColor;
-    }
-    if (rgba[4]) {
-        return cssColor;
-    }
-    var hex = ColorpickerDialog.convertRgbToHex(
-        parseInt(rgba[1]),
-        parseInt(rgba[2]),
-        parseInt(rgba[3])
-    );
-    if (!hex) {
-        return cssColor; // TODO handle error
-    }
-    return hex.hex.toUpperCase();
-}
+var applyColor = function (ev, eventName, color) {
+    var layoutInfo = dom.makeLayoutInfo(ev.data.target);
+    $.summernote.pluginEvents[eventName](undefined, eventHandler.modules.editor, layoutInfo, color || ev.data.cssColor);
+};
 
 // Update and change the popovers content, and add history button
-var fn_createPalette = renderer.createPalette;
-renderer.createPalette = function ($container, options) {
-    fn_createPalette.call(this, $container, options);
-
-    if (!QWeb.has_template('web_editor.colorpicker')) {
-        return;
-    }
-
-    var $clpicker = $(QWeb.render('web_editor.colorpicker'));
-
-    var groups;
-    if ($clpicker.is("colorpicker")) {
-        groups = _.map($clpicker.find('[data-name="theme"]'), function (el) {
-            return $(el).find("button").empty();
+renderer.createPalette = async function ($container, options) {
+    const $dropdownContent = $container.find(".colorPalette");
+    _.each($dropdownContent, elem => {
+        const eventName = elem.dataset.eventName;
+        const colorpicker = new ColorPaletteWidget(undefined, {
```

Comment:
Better `null` than `undefined` but is a parent not necessary for the widget to work (and even if no, no magical/hacky way to have a parent ?).